cmd/ormb-storage-initializer/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright © 2020 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/caicloud/ormb/pkg/consts"
+	"github.com/caicloud/ormb/pkg/oci"
+	"github.com/caicloud/ormb/pkg/ormb"
+)
+
+var ormbClient ormb.ORMB
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:     "ormb-storage-initializer",
+	Short:   "Download model from remote registry in Seldon Core and KFSerivng",
+	Long:    ``,
+	PreRunE: preRunE,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		modelURI := args[0]
+		dstDir := args[1]
+
+		// Get username and password from environment
+		// Here AWS_SECRET_ACCESS_KEY and AWS_ACCESS_KEY_ID are used
+		// because Seldon Core does not support renaming the environment variable name.
+		username := viper.GetString("AWS_ACCESS_KEY_ID")
+		pwd := viper.GetString("AWS_SECRET_ACCESS_KEY")
+		// Get the host from the URL.
+		strs := strings.Split(modelURI, "/")
+		if len(strs) == 0 {
+			return fmt.Errorf("Failed to get the host from %s", modelURI)
+		}
+		fmt.Printf("Logging to the remote registry %s\n", strs[0])
+		fmt.Printf("Username: %s\n", username)
+		if err := ormbClient.Login(strs[0], username, pwd, true); err != nil {
```

Comment:
Why it need login before `ormb.NewOCIORMB`,  I think if `ormbClient.Login` between `ormb.NewOCIORMB` and ormbClient.Pull,   it is more reasonable.  And `PreRunE` can be deleted.

Reply:
We need two calls for ormb.NewOCIORMB to let it know the credentials.

Reply:
OK, If it is right, LGTM, I'am not understand it now, let me think it carefully.

Reply:
BTW, we only have one command in storage initializer, then we do not actually need the prerun here since it does not benefit us by code reusing